javascript/URDFLoader.js

```diff
@@ -143,19 +150,33 @@ class URDFLoader {
     }
 
     /* Private Functions */
+    // Resolves the path of mesh files
+    _resolveMeshPath(meshPath){
+        const filename = meshPath.replace(/^package:\/\//, '');
+        const [pkgName, relPath] = filename.split(/\/(.+)/);
+
+        if (pkgName in this.pkgMap){
+            return this.pkgMap[pkgName] + '/' + relPath;
+        } else {
+            console.warn("Error: Pkg ", pkgName, " not found in pkg list! Default is first pkg.");
+            // Use the first pkg by default and add the filename
+            return Object.values(this.pkgMap)[0] + '/' + filename;
```

Comment:
The error will get fired even when you just provide a single package via string, right? Maybe we can provide a default path via a wildcard-like notation, such as `{ '*': '.../default_package' }`. That way we'll only print an error when when a package hasn't been provided by any of the expected means.

Reply:
The error will not get fired if only one package is provided. This works fine.

It will however get fired for your urdf example because of:

> <filename="package://T12/meshes/Body.STL" />

The problem here is that the mesh-path does not start from a ROS package (how it would typically be the case in ROS) but from a relative path within the hosted folder. It cannot lookup the package "T12" in the provided object because the package path does not end by the folder "T12" but by the parent folder "urdf". Since the lookup fails the error is fired.

I handled this exception by merging the package path "..../urdf" with the mesh path "T12..../" to solve the describes problem.

Reply:
That's fair. I suppose the "default package" approach is technically incorrect by the ROS ecosystem. I'll think about this a bit more.